sdk/cpp/core/src/gnmi_client.cpp

```diff
@@ -0,0 +1,273 @@
+#include <iostream>
+#include <fstream>
+#include <memory>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <iterator>
+
+#include "errors.hpp"
+#include "gnmi_client.hpp"
+#include "types.hpp"
+#include "logger.hpp"
+#include "path_api.hpp"
+#include "json.hpp"
+
+#include <libyang/libyang.h>
+#include <grpc++/grpc++.h>
+#include <grpc++/client_context.h>
+#include <grpc++/create_channel.h>
+#include <libgnmi/gnmi.grpc.pb.h>
+#include <libgnmi/gnmi.pb.h>
+#include "entity_data_node_walker.hpp"
+#include "ietf_parser.hpp"
+#include "ydk_yang.hpp"
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+using gnmi::gNMI;
+using gnmi::CapabilityRequest;
+using gnmi::CapabilityResponse;
+using gnmi::GetRequest;
+using gnmi::GetResponse;
+
+using namespace std;
+using namespace ydk;
+using json = nlohmann::json;
+
+typedef vector<string> StringVec;
+std::ifstream capfile("/usr/local/share/ydk/0.0.0.0\:50051/capabilities.txt");
```

Comment:
Is the capability exchange process works similar as NETCONF? Why is a local file(and port number is added in the path) is used?